lib/services/nodes-api-service.js

```diff
@@ -249,18 +249,36 @@ function nodeApiServiceFactory(
         return Promise.try(function() {
             query = waterline.nodes.find(query);
 
-            if (options.skip) query.skip(options.skip);
-            if (options.limit) query.limit(options.limit);
+            if (options.skip) {
+                query.skip(options.skip);
+            }
+            if (options.limit) {
+                query.limit(options.limit);
+            }
 
             return query.populate('obms');
         });
     };
 
     NodeApiService.prototype.postNode = function(body) {
-        return Promise.resolve().then(function() {
+        return Promise.try (function() {
+            assert.string(body.type, 'node type is required');
```

Comment:
@yyscamper This should be handled in the schema; we should be adding `type` to the list of required properties: https://github.com/RackHD/on-http/blob/master/static/schemas/2.0/node.2.0.json#L54.  You shouldn't need to validate the type since we have a enum in the schema with all node types.  The new `obms` request option should be added to the nodes schema as an array of objects.  The tricky part here is that we should be referencing the obm schema (https://github.com/RackHD/on-http/tree/master/static/schemas/obms) here to validate the obms themselves.